pyresample/boundary.py

```diff
@@ -54,22 +54,156 @@ def draw(self, mapper, options, **more_options):
         self.contour_poly.draw(mapper, options, **more_options)
 
 
+def _is_corner_is_clockwise(lon1, lat1, corner_lon, corner_lat, lon2, lat2):
+    """Determine if coordinates follow a clockwise path.
+
+    This uses :class:`pyresample.spherical.Arc` to determine the angle
+    between the first line segment (Arc) from (lon1, lat1) to
+    (corner_lon, corner_lat) and the second line segment from
+    (corner_lon, corner_lat) to (lon2, lat2). A straight line would
+    produce an angle of 0, a clockwise path would have a negative angle,
+    and a counter-clockwise path would have a positive angle.
+
+    """
+    import math
+
+    from pyresample.spherical import Arc, SCoordinate
+    point1 = SCoordinate(math.radians(lon1), math.radians(lat1))
+    point2 = SCoordinate(math.radians(corner_lon), math.radians(corner_lat))
+    point3 = SCoordinate(math.radians(lon2), math.radians(lat2))
+    arc1 = Arc(point1, point2)
+    arc2 = Arc(point2, point3)
+    angle = arc1.angle(arc2)
+    is_clockwise = -np.pi < angle < 0
+    return is_clockwise
+
+
+def _is_boundary_clockwise(sides_lons, sides_lats):
+    """Determine if the boundary sides are clockwise."""
+    is_clockwise = _is_corner_is_clockwise(
+        lon1=sides_lons[0][-2],
+        lat1=sides_lats[0][-2],
+        corner_lon=sides_lons[0][-1],
+        corner_lat=sides_lats[0][-1],
+        lon2=sides_lons[1][1],
+        lat2=sides_lats[1][1])
+    return is_clockwise
+
+
+def _check_sides_list(sides):
+    if not isinstance(sides, list):
+        raise TypeError("Boundary sides must be a list")
```

Comment:
My gut reaction is that this shouldn't be added and instead put `sides: list` or even better `sides: Iterable` depending on how it needs to be used...but even better still would be to put this annotation where the sides are passed (`__init__`)? This would put the checks on the static mypy side of things rather than at run time. If something is passing the wrong type of arguments then mypy should complain about it.

Oh you could maybe do `tuple[float, float, float, float]` or whatever the type is. That would *force* it to be a tuple but should also define that it needs to have 4 sides.

Reply:
Eh maybe the `Sequence` type annotation is what you should use. You could keep the length check, but something in me wants it to be removed and let it fail later when it is actually used.

Reply:
Outdated. I created the `BoundarySides` object to very the object format